doc/build_ref/TrilinosBuildReferenceTemplate.rst

```diff
@@ -105,3 +105,65 @@ In order do co-development of TriBTS and Trilinos (see http://http://trac.trilin
 
 (NOTE: You have to use the data-type ``STRING`` with ``Trilinos_TRIBITS_DIR``
 or CMake will automatically assume it is relative to the build dir!)
+
+
+Example: Configuring with Kokkos and advanced back-ends
+--------------------------------------------------------
+
+Kokkos serves as a basis for many advanced trilinos packages, especially those
```

Comment:
This doesn't say what Kokkos _is_ or why users might want it.  I'm also not a fan of words like "advanced" that don't have a clear meaning.  It's like saying "fastest" or "only the latest."  Kokkos will work fine on older hardware too.

How about this: "Kokkos (https://github.com/kokkos/kokkos) is a C++ implementation of a cross-platform shared-memory parallel programming model.  Many Trilinos packages, and other stand-alone applications, use it to implement parallel algorithms.  Kokkos has two different build systems: Trilinos' CMake-based build system, and its own separate build system."

Regarding "...and this build system has more options than that supported by Trilinos by default..." -- isn't the point of this pull request to fix that?

Reply:
I took your verbiage.  Thanks!

My goal was to explain why a user would have something like `-DTPL_ENABLE_CUDA=True -DKOKKOS_ENABLE_CUDA_LDG_INTRINSIC=True`.  Why TPL for one and not the other? 

I think with your fixes it is sufficient.